physicsnemo/models/diffusion/layers.py

```
@@ -19,15 +19,23 @@
 Diffusion-Based Generative Models".
 """
 
+import contextlib
 from typing import Any, Dict, List
 
 import numpy as np
+import nvtx
 import torch
+import torch.cuda.amp as amp
+from apex.contrib.group_norm import GroupNorm as ApexGroupNorm
```

@jialusui1102 Could you make the apex imports more robust with something like:

```python
is_apex_available = False
if torch.cuda.is_available():
    try:
        apex_gn = getattr(
            importlib.import_module("apex.contrib_group_norm"), "FusedAdam"
        )
        is_apex_available = True
    except ImportError:
        pass
```

Also we should raise an error when `use_apex_gn = True` and `is_apex_available` is `False`

---

**EDIT:** done, I directly committed the change to your PR